reducer/src/main/java/com/graphicsfuzz/reducer/glslreducers/EliminateInjectionMacrosVisitor.java

```diff
@@ -129,15 +165,58 @@ private void cleanUpMacros(Expr parent) {
   @Override
   public void visitScalarInitializer(ScalarInitializer scalarInitializer) {
     super.visitScalarInitializer(scalarInitializer);
-    if (MacroNames.isFuzzed(scalarInitializer.getExpr())) {
+    if (MacroNames.isIdentity(scalarInitializer.getExpr())
+        || MacroNames.isZero(scalarInitializer.getExpr())
+        || MacroNames.isOne(scalarInitializer.getExpr())
+        || MacroNames.isFalse(scalarInitializer.getExpr())
+        || MacroNames.isTrue(scalarInitializer.getExpr())) {
+      scalarInitializer.setExpr(scalarInitializer.getExpr().getChild(1));
+    } else if (MacroNames.isFuzzed(scalarInitializer.getExpr())
+        || MacroNames.isDeadByConstruction(scalarInitializer.getExpr())) {
       scalarInitializer.setExpr(scalarInitializer.getExpr().getChild(0));
     }
   }
 
   private void replaceChildWithGrandchild(Expr parent, int childIndex, int grandchildIndex) {
     assert parent.getChild(childIndex).getNumChildren() > grandchildIndex;
-    parent
-        .setChild(childIndex, new ParenExpr(parent.getChild(childIndex).getChild(grandchildIndex)));
+
+    if (suitableToRemoveParentheses(parent, parent.getChild(childIndex))) {
```

Comment:
Rather than deciding whether to remove parentheses, we are in fact deciding whether to add them.  So please can you rename the "suitableToRemoveParentheses" method along those lines?  It should check whether it is necessary to add them.

Reply:
Fixed